csharp/src/Apache.Arrow.Flight/Ticket.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Google.Protobuf;
+
+namespace Apache.Arrow.Flight
+{
+    public class Ticket
+    {
+        private readonly Protocol.Ticket _ticket;
+        internal Ticket(Protocol.Ticket ticket)
+        {
+            _ticket = ticket;
+        }
+
+        public Ticket(ByteString ticket)
+        {
+            _ticket = new Protocol.Ticket()
+            {
+                Ticket_ = ticket
+            };
+        }
+
+        public Ticket(string ticket)
+            : this(ByteString.CopyFromUtf8(ticket))
+        {
+        }
+
+        public Ticket(byte[] bytes)
+            : this(ByteString.CopyFrom(bytes))
+        {
+        }
+
+        public string TicketString => _ticket.Ticket_.ToStringUtf8();
+
+        public ByteString TicketByteString => _ticket.Ticket_;
+
+        public byte[] TicketBytes => _ticket.Ticket_.ToByteArray();
```

Comment:
Do we really need all 3 of these as public properties?

One concern I have is that properties are supposed to be cheap, but every time you call `TicketBytes` or `TicketString`, it allocates new memory.

Can we get away with just exposing `TicketByteString` publicaly? And callers can turn it into a `byte[]` or `string` as necessary.

Reply:
Absolutely, great feedback, thanks 👍

Reply:
Note, this apply elsewhere where we have the same 3 public properties. Ex. `FlightDescriptor.Command`

Reply:
Removed the 3 properties from all classes they existed in and they now only expose ByteString.